recipes-extended/xen/files/libxl-fix-flr.patch

```diff
@@ -0,0 +1,154 @@
+Index: xen-4.6.6/tools/libxl/libxl.c
+===================================================================
+--- xen-4.6.6.orig/tools/libxl/libxl.c
++++ xen-4.6.6/tools/libxl/libxl.c
+@@ -1604,7 +1604,13 @@ static void domain_destroy_callback(libx
+         LOG(ERROR, "unable to destroy guest with domid %u", dis->domid);
+         dds->rc = rc;
+     }
+-
++    if(dis->pciw) {
++        for (int i = 0; i < dis->pciw->num_devs; i++) {
++            libxl__device_pci_reset(gc, dis->pciw->pcidevs[i].domain, dis->pciw->pcidevs[i].bus, dis->pciw->pcidevs[i].dev, dis->pciw->pcidevs[i].func);
++        } 
++        free(dis->pciw->pcidevs);
++        free(dis->pciw);
++    }
+     dds->domain_finished = 1;
+     destroy_finish_check(egc, dds);
+ }
+@@ -1672,8 +1678,8 @@ void libxl__destroy_domid(libxl__egc *eg
+         goto out;
+     }
+ 
+-    if (libxl__device_pci_destroy_all(gc, domid) < 0)
+-        LIBXL__LOG(ctx, LIBXL__LOG_ERROR, "pci shutdown failed for domid %d", domid);
++    libxl_pci_dev_wrap *pciw = libxl__device_pci_destroy_all(gc, domid);
++    dis->pciw = pciw;
```

Comment:
Another stylistic comment. Why use the extra variable to temporarily store the return value and not just store the return value directly into struct instance?

Reply:
Can't remember specifically, might have planned on using _pciw_ later on in the function, never did, and then never performed the direct assignment.